module/zfs/zvol.c

```diff
@@ -1750,3 +1750,52 @@ zvol_fini_impl(void)
 	list_destroy(&zvol_state_list);
 	rw_destroy(&zvol_state_lock);
 }
+
+/* ZFS ZVOLDI */
+_Function_class_(PINTERFACE_REFERENCE)
+void IncZvolRef(PVOID Context) {
+	zvol_state_t* zv = (zvol_state_t*)Context;
+	mutex_enter(&zv->zv_state_lock);
+	atomic_inc_32(&zv->zv_open_count);
+	mutex_exit(&zv->zv_state_lock);
```

Comment:
mutex around an atomic seems quite redundant?

Reply:
Yes, removed the mutex lock.